chains/evm/contracts/poolsV2/TokenPool.sol

```diff
@@ -42,11 +90,193 @@ abstract contract TokenPool is IPoolV2, TokenPoolV1 {
+  /// @param finality The finality depth requested by the message. A value of zero uses the standard lane.
+  /// @return fastTransferFeeBps The fee in basis points for fast transfers, or zero if standard transfer is used.
+  function _validateLockOrBurn(
+    Pool.LockOrBurnInV1 memory lockOrBurnIn,
+    uint16 finality
+  ) internal returns (uint16 fastTransferFeeBps) {
+    if (!isSupportedToken(lockOrBurnIn.localToken)) revert InvalidToken(lockOrBurnIn.localToken);
+    if (IRMN(i_rmnProxy).isCursed(bytes16(uint128(lockOrBurnIn.remoteChainSelector)))) revert CursedByRMN();
+    _checkAllowList(lockOrBurnIn.originalSender);
+
+    _onlyOnRamp(lockOrBurnIn.remoteChainSelector);
+    FastFinalityConfig storage finalityConfig = s_finalityConfig;
+    uint256 amount = lockOrBurnIn.amount;
+    if (finality != 0 && finalityConfig.finalityThreshold != 0) {
+      if (finality < finalityConfig.finalityThreshold) {
+        revert InvalidFinality(finality, finalityConfig.finalityThreshold);
+      }
+      if (amount > finalityConfig.maxAmountPerRequest) {
+        revert AmountExceedsMaxPerRequest(amount, finalityConfig.maxAmountPerRequest);
+      }
+
+      finalityConfig.outboundRateLimiterConfig[lockOrBurnIn.remoteChainSelector]._consume(
+        amount, lockOrBurnIn.localToken
+      );
+      emit FastTransferOutboundRateLimitConsumed(lockOrBurnIn.remoteChainSelector, lockOrBurnIn.localToken, amount);
+      return finalityConfig.fastTransferFeeBps;
+    } else {
+      _consumeOutboundRateLimit(lockOrBurnIn.remoteChainSelector, amount);
+    }
+
+    return 0;
+  }
+
+  /// @notice Validates a release or mint request and enforces the appropriate inbound rate limits.
+  /// @dev The validation checks token support, RMN curse status, offRamp access, remote pool configuration,
+  /// finality requirements, and consumes either the fast-transfer inbound bucket or the standard bucket.
+  /// @param releaseOrMintIn The input to validate. The remote chain, pool, and token must all be configured.
+  /// @param localAmount The amount to release or mint on the local chain after any decimal conversion.
+  /// @param finality The finality depth requested by the message. A value of zero uses the standard lane.
+  function _validateReleaseOrMint(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn,
+    uint256 localAmount,
+    uint16 finality
+  ) internal {
+    if (!isSupportedToken(releaseOrMintIn.localToken)) revert InvalidToken(releaseOrMintIn.localToken);
+    if (IRMN(i_rmnProxy).isCursed(bytes16(uint128(releaseOrMintIn.remoteChainSelector)))) revert CursedByRMN();
+    _onlyOffRamp(releaseOrMintIn.remoteChainSelector);
+
+    // Validates that the source pool address is configured on this pool.
+    if (!isRemotePool(releaseOrMintIn.remoteChainSelector, releaseOrMintIn.sourcePoolAddress)) {
+      revert InvalidSourcePoolAddress(releaseOrMintIn.sourcePoolAddress);
+    }
+
+    FastFinalityConfig storage finalityConfig = s_finalityConfig;
```

Comment:
You can assign this within the `if`, or even better, not assign at all and just reference directly